src/lib/core/CHIPPersistentStorageDelegate.h

```diff
@@ -124,15 +53,15 @@ class DLL_EXPORT PersistentStorageDelegate
      * @param[in] value Value to be set
      * @param[in] size Size of the Value
      */
-    virtual CHIP_ERROR SyncSetKeyValue(const char * key, const void * value, uint16_t size) { return CHIP_ERROR_NOT_IMPLEMENTED; }
+    virtual CHIP_ERROR SyncSetKeyValue(const char * key, const void * value, uint16_t size) = 0;
```

Comment:
I almost wonder whether using `ByteSpan` here would be more idiomatic.... Followup to think about that?

Reply:
I think it would be better, however I would like to update this once the api is finished moving to KVS. I believe kvs does not use span because it was not a thing when it was included (and we use a C++ standard that is old enough to not include it).

Reply:
Yes, I am very much not asking for the span thing to happen in this PR!